File: google/cloud/bigquery/v2/google-cloud-bigquery-v2-ruby/test/google/cloud/bigquery/v2/model_service_client_test.rb
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

require "minitest/autorun"
require "minitest/spec"

require "google/gax"

require "google/cloud/bigquery"
require "google/cloud/bigquery/v2/model_service_client"
require "google/cloud/bigquery/v2/model_services_pb"

class CustomTestError_v2 < StandardError; end

# Mock for the GRPC::ClientStub class.
class MockGrpcClientStub_v2

  # @param expected_symbol [Symbol] the symbol of the grpc method to be mocked.
  # @param mock_method [Proc] The method that is being mocked.
  def initialize(expected_symbol, mock_method)
    @expected_symbol = expected_symbol
    @mock_method = mock_method
  end

  # This overrides the Object#method method to return the mocked method when the mocked method
  # is being requested. For methods that aren't being tested, this method returns a proc that
  # will raise an error when called. This is to assure that only the mocked grpc method is being
  # called.
  #
  # @param symbol [Symbol] The symbol of the method being requested.
  # @return [Proc] The proc of the requested method. If the requested method is not being mocked
  #   the proc returned will raise when called.
  def method(symbol)
    return @mock_method if symbol == @expected_symbol

    # The requested method is not being tested, raise if it called.
    proc do
      raise "The method #{symbol} was unexpectedly called during the " \
        "test for #{@expected_symbol}."
    end
  end
end

class MockModelServiceCredentials_v2 < Google::Cloud::Bigquery::V2::Credentials
  def initialize(method_name)
    @method_name = method_name
  end

  def updater_proc
    proc do
      raise "The method `#{@method_name}` was trying to make a grpc request. This should not " \
          "happen since the grpc layer is being mocked."
    end
  end
end

describe Google::Cloud::Bigquery::V2::ModelServiceClient do

  describe 'get_model' do
    custom_error = CustomTestError_v2.new "Custom test error for Google::Cloud::Bigquery::V2::ModelServiceClient#get_model."

    it 'invokes get_model without error' do
      # Create request parameters
      project_id = ''
      dataset_id = ''
      model_id = ''

      # Create expected grpc response
      etag = "etag3123477"
      creation_time = 1932333101
      last_modified_time = 671513446
      description = "description-1724546052"
      friendly_name = "friendlyName1451097503"
      expiration_time = 767170141
      location = "location1901043637"
      expected_response = {
        etag: etag,
        creation_time: creation_time,
        last_modified_time: last_modified_time,
        description: description,
        friendly_name: friendly_name,
        expiration_time: expiration_time,
        location: location
      }
      expected_response = Google::Gax::to_proto(expected_response, Google::Cloud::Bigquery::V2::Model)

      # Mock Grpc layer
      mock_method = proc do |request|
        assert_instance_of(Google::Cloud::Bigquery::V2::GetModelRequest, request)
        assert_equal(project_id, request.project_id)
        assert_equal(dataset_id, request.dataset_id)
        assert_equal(model_id, request.model_id)
        OpenStruct.new(execute: expected_response)
      end
      mock_stub = MockGrpcClientStub_v2.new(:get_model, mock_method)

      # Mock auth layer
      mock_credentials = MockModelServiceCredentials_v2.new("get_model")

      Google::Cloud::Bigquery::V2::ModelService::Stub.stub(:new, mock_stub) do
        Google::Cloud::Bigquery::V2::Credentials.stub(:default, mock_credentials) do
          client = Google::Cloud::Bigquery.new(version: :v2)

          # Call method
          response = client.get_model(
            project_id,
            dataset_id,
            model_id
          )

          # Verify the response
          assert_equal(expected_response, response)

          # Call method with block
          client.get_model(
            project_id,
            dataset_id,
            model_id
          ) do |response, operation|
            # Verify the response
            assert_equal(expected_response, response)
            refute_nil(operation)
          end
        end
      end
    end

    it 'invokes get_model with error' do
      # Create request parameters
      project_id = ''
      dataset_id = ''
      model_id = ''

      # Mock Grpc layer
      mock_method = proc do |request|
        assert_instance_of(Google::Cloud::Bigquery::V2::GetModelRequest, request)
        assert_equal(project_id, request.project_id)
        assert_equal(dataset_id, request.dataset_id)
        assert_equal(model_id, request.model_id)
        raise custom_error
      end
      mock_stub = MockGrpcClientStub_v2.new(:get_model, mock_method)

      # Mock auth layer
      mock_credentials = MockModelServiceCredentials_v2.new("get_model")

      Google::Cloud::Bigquery::V2::ModelService::Stub.stub(:new, mock_stub) do
        Google::Cloud::Bigquery::V2::Credentials.stub(:default, mock_credentials) do
          client = Google::Cloud::Bigquery.new(version: :v2)

          # Call method
          err = assert_raises Google::Gax::GaxError, CustomTestError_v2 do
            client.get_model(
              project_id,
              dataset_id,
              model_id
            )
          end

          # Verify the GaxError wrapped the custom error that was raised.
          assert_match(custom_error.message, err.message)
        end
      end
    end
  end

  describe 'list_models' do
    custom_error = CustomTestError_v2.new "Custom test error for Google::Cloud::Bigquery::V2::ModelServiceClient#list_models."

    it 'invokes list_models without error' do
      # Create request parameters
      project_id = ''
      dataset_id = ''

      # Create expected grpc response
      next_page_token = "nextPageToken-1530815211"
      expected_response = { next_page_token: next_page_token }
      expected_response = Google::Gax::to_proto(expected_response, Google::Cloud::Bigquery::V2::ListModelsResponse)

      # Mock Grpc layer
      mock_method = proc do |request|
        assert_instance_of(Google::Cloud::Bigquery::V2::ListModelsRequest, request)
        assert_equal(project_id, request.project_id)
        assert_equal(dataset_id, request.dataset_id)
        OpenStruct.new(execute: expected_response)
      end
      mock_stub = MockGrpcClientStub_v2.new(:list_models, mock_method)

      # Mock auth layer
      mock_credentials = MockModelServiceCredentials_v2.new("list_models")

      Google::Cloud::Bigquery::V2::ModelService::Stub.stub(:new, mock_stub) do
        Google::Cloud::Bigquery::V2::Credentials.stub(:default, mock_credentials) do
          client = Google::Cloud::Bigquery.new(version: :v2)

          # Call method
          response = client.list_models(project_id, dataset_id)

          # Verify the response
          assert_equal(expected_response, response)

          # Call method with block
          client.list_models(project_id, dataset_id) do |response, operation|
            # Verify the response
            assert_equal(expected_response, response)
            refute_nil(operation)
          end
        end
      end
    end

    it 'invokes list_models with error' do
      # Create request parameters
      project_id = ''
      dataset_id = ''

      # Mock Grpc layer
      mock_method = proc do |request|
        assert_instance_of(Google::Cloud::Bigquery::V2::ListModelsRequest, request)
        assert_equal(project_id, request.project_id)
        assert_equal(dataset_id, request.dataset_id)
        raise custom_error
      end
      mock_stub = MockGrpcClientStub_v2.new(:list_models, mock_method)

      # Mock auth layer
      mock_credentials = MockModelServiceCredentials_v2.new("list_models")

      Google::Cloud::Bigquery::V2::ModelService::Stub.stub(:new, mock_stub) do
        Google::Cloud::Bigquery::V2::Credentials.stub(:default, mock_credentials) do
          client = Google::Cloud::Bigquery.new(version: :v2)

          # Call method
          err = assert_raises Google::Gax::GaxError, CustomTestError_v2 do
            client.list_models(project_id, dataset_id)
          end

          # Verify the GaxError wrapped the custom error that was raised.
          assert_match(custom_error.message, err.message)
        end
      end
    end
  end

  describe 'patch_model' do
    custom_error = CustomTestError_v2.new "Custom test error for Google::Cloud::Bigquery::V2::ModelServiceClient#patch_model."

    it 'invokes patch_model without error' do
      # Create request parameters
      project_id = ''
      dataset_id = ''
      model_id = ''
      model = {}

      # Create expected grpc response
      etag = "etag3123477"
      creation_time = 1932333101
      last_modified_time = 671513446
      description = "description-1724546052"
      friendly_name = "friendlyName1451097503"
      expiration_time = 767170141
      location = "location1901043637"
      expected_response = {
        etag: etag,
        creation_time: creation_time,
        last_modified_time: last_modified_time,
        description: description,
        friendly_name: friendly_name,
        expiration_time: expiration_time,
        location: location
      }
      expected_response = Google::Gax::to_proto(expected_response, Google::Cloud::Bigquery::V2::Model)

      # Mock Grpc layer
      mock_method = proc do |request|
        assert_instance_of(Google::Cloud::Bigquery::V2::PatchModelRequest, request)
        assert_equal(project_id, request.project_id)
        assert_equal(dataset_id, request.dataset_id)
        assert_equal(model_id, request.model_id)
        assert_equal(Google::Gax::to_proto(model, Google::Cloud::Bigquery::V2::Model), request.model)
        OpenStruct.new(execute: expected_response)
      end
      mock_stub = MockGrpcClientStub_v2.new(:patch_model, mock_method)

      # Mock auth layer
      mock_credentials = MockModelServiceCredentials_v2.new("patch_model")

      Google::Cloud::Bigquery::V2::ModelService::Stub.stub(:new, mock_stub) do
        Google::Cloud::Bigquery::V2::Credentials.stub(:default, mock_credentials) do
          client = Google::Cloud::Bigquery.new(version: :v2)

          # Call method
          response = client.patch_model(
            project_id,
            dataset_id,
            model_id,
            model
          )

          # Verify the response
          assert_equal(expected_response, response)

          # Call method with block
          client.patch_model(
            project_id,
            dataset_id,
            model_id,
            model
          ) do |response, operation|
            # Verify the response
            assert_equal(expected_response, response)
            refute_nil(operation)
          end
        end
      end
    end

    it 'invokes patch_model with error' do
      # Create request parameters
      project_id = ''
      dataset_id = ''
      model_id = ''
      model = {}

      # Mock Grpc layer
      mock_method = proc do |request|
        assert_instance_of(Google::Cloud::Bigquery::V2::PatchModelRequest, request)
        assert_equal(project_id, request.project_id)
        assert_equal(dataset_id, request.dataset_id)
        assert_equal(model_id, request.model_id)
        assert_equal(Google::Gax::to_proto(model, Google::Cloud::Bigquery::V2::Model), request.model)
        raise custom_error
      end
      mock_stub = MockGrpcClientStub_v2.new(:patch_model, mock_method)

      # Mock auth layer
      mock_credentials = MockModelServiceCredentials_v2.new("patch_model")

      Google::Cloud::Bigquery::V2::ModelService::Stub.stub(:new, mock_stub) do
        Google::Cloud::Bigquery::V2::Credentials.stub(:default, mock_credentials) do
          client = Google::Cloud::Bigquery.new(version: :v2)

          # Call method
          err = assert_raises Google::Gax::GaxError, CustomTestError_v2 do
            client.patch_model(
              project_id,
              dataset_id,
              model_id,
              model
            )
          end

          # Verify the GaxError wrapped the custom error that was raised.
          assert_match(custom_error.message, err.message)
        end
      end
    end
  end

  describe 'delete_model' do
    custom_error = CustomTestError_v2.new "Custom test error for Google::Cloud::Bigquery::V2::ModelServiceClient#delete_model."

    it 'invokes delete_model without error' do
      # Create request parameters
      project_id = ''
      dataset_id = ''
      model_id = ''

      # Mock Grpc layer
      mock_method = proc do |request|
        assert_instance_of(Google::Cloud::Bigquery::V2::DeleteModelRequest, request)
        assert_equal(project_id, request.project_id)
        assert_equal(dataset_id, request.dataset_id)
        assert_equal(model_id, request.model_id)
        OpenStruct.new(execute: nil)
      end
      mock_stub = MockGrpcClientStub_v2.new(:delete_model, mock_method)

      # Mock auth layer
      mock_credentials = MockModelServiceCredentials_v2.new("delete_model")

      Google::Cloud::Bigquery::V2::ModelService::Stub.stub(:new, mock_stub) do
        Google::Cloud::Bigquery::V2::Credentials.stub(:default, mock_credentials) do
          client = Google::Cloud::Bigquery.new(version: :v2)

          # Call method
          response = client.delete_model(
            project_id,
            dataset_id,
            model_id
          )

          # Verify the response
          assert_nil(response)

          # Call method with block
          client.delete_model(
            project_id,
            dataset_id,
            model_id
          ) do |response, operation|
            # Verify the response
            assert_nil(response)
            refute_nil(operation)
          end
        end
      end
    end

    it 'invokes delete_model with error' do
      # Create request parameters
      project_id = ''
      dataset_id = ''
      model_id = ''

      # Mock Grpc layer
      mock_method = proc do |request|
        assert_instance_of(Google::Cloud::Bigquery::V2::DeleteModelRequest, request)
        assert_equal(project_id, request.project_id)
        assert_equal(dataset_id, request.dataset_id)
        assert_equal(model_id, request.model_id)
        raise custom_error
      end
      mock_stub = MockGrpcClientStub_v2.new(:delete_model, mock_method)

      # Mock auth layer
      mock_credentials = MockModelServiceCredentials_v2.new("delete_model")

      Google::Cloud::Bigquery::V2::ModelService::Stub.stub(:new, mock_stub) do
        Google::Cloud::Bigquery::V2::Credentials.stub(:default, mock_credentials) do
          client = Google::Cloud::Bigquery.new(version: :v2)

          # Call method
          err = assert_raises Google::Gax::GaxError, CustomTestError_v2 do
            client.delete_model(
              project_id,
              dataset_id,
              model_id
            )
          end

          # Verify the GaxError wrapped the custom error that was raised.
          assert_match(custom_error.message, err.message)
        end
      end
    end
  end
end

File: google/cloud/bigquery/v2/google-cloud-bigquery-v2-java/gapic-google-cloud-bigquery-v2-java/src/test/java/com/google/cloud/bigquery/v2/ModelServiceClientTest.java
/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.google.cloud.bigquery.v2;

import com.google.api.gax.core.NoCredentialsProvider;
import com.google.api.gax.grpc.GaxGrpcProperties;
import com.google.api.gax.grpc.testing.LocalChannelProvider;
import com.google.api.gax.grpc.testing.MockGrpcService;
import com.google.api.gax.grpc.testing.MockServiceHelper;
import com.google.api.gax.rpc.ApiClientHeaderProvider;
import com.google.api.gax.rpc.InvalidArgumentException;
import com.google.protobuf.AbstractMessage;
import com.google.protobuf.Empty;
import com.google.protobuf.UInt32Value;
import io.grpc.StatusRuntimeException;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.UUID;
import javax.annotation.Generated;
import org.junit.After;
import org.junit.AfterClass;
import org.junit.Assert;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;

@Generated("by gapic-generator-java")
public class ModelServiceClientTest {
  private static MockModelService mockModelService;
  private static MockServiceHelper mockServiceHelper;
  private LocalChannelProvider channelProvider;
  private ModelServiceClient client;

  @BeforeClass
  public static void startStaticServer() {
    mockModelService = new MockModelService();
    mockServiceHelper =
        new MockServiceHelper(
            UUID.randomUUID().toString(), Arrays.<MockGrpcService>asList(mockModelService));
    mockServiceHelper.start();
  }

  @AfterClass
  public static void stopServer() {
    mockServiceHelper.stop();
  }

  @Before
  public void setUp() throws IOException {
    mockServiceHelper.reset();
    channelProvider = mockServiceHelper.createChannelProvider();
    ModelServiceSettings settings =
        ModelServiceSettings.newBuilder()
            .setTransportChannelProvider(channelProvider)
            .setCredentialsProvider(NoCredentialsProvider.create())
            .build();
    client = ModelServiceClient.create(settings);
  }

  @After
  public void tearDown() throws Exception {
    client.close();
  }

  @Test
  public void getModelTest() throws Exception {
    ModelProto.Model expectedResponse =
        ModelProto.Model.newBuilder()
            .setEtag("etag3123477")
            .setModelReference(ModelReferenceProto.ModelReference.newBuilder().build())
            .setCreationTime(1932333101)
            .setLastModifiedTime(-671513446)
            .setDescription("description-1724546052")
            .setFriendlyName("friendlyName461933014")
            .putAllLabels(new HashMap<String, String>())
            .setExpirationTime(767170141)
            .setLocation("location1901043637")
            .setEncryptionConfiguration(
                EncryptionConfigProto.EncryptionConfiguration.newBuilder().build())
            .addAllTrainingRuns(new ArrayList<ModelProto.Model.TrainingRun>())
            .addAllFeatureColumns(new ArrayList<StandardSqlProto.StandardSqlField>())
            .addAllLabelColumns(new ArrayList<StandardSqlProto.StandardSqlField>())
            .build();
    mockModelService.addResponse(expectedResponse);

    String projectId = "projectId-894832108";
    String datasetId = "datasetId-345342029";
    String modelId = "modelId1226956324";

    ModelProto.Model actualResponse = client.getModel(projectId, datasetId, modelId);
    Assert.assertEquals(expectedResponse, actualResponse);

    List<AbstractMessage> actualRequests = mockModelService.getRequests();
    Assert.assertEquals(1, actualRequests.size());
    ModelProto.GetModelRequest actualRequest = ((ModelProto.GetModelRequest) actualRequests.get(0));

    Assert.assertEquals(projectId, actualRequest.getProjectId());
    Assert.assertEquals(datasetId, actualRequest.getDatasetId());
    Assert.assertEquals(modelId, actualRequest.getModelId());
    Assert.assertTrue(
        channelProvider.isHeaderSent(
            ApiClientHeaderProvider.getDefaultApiClientHeaderKey(),
            GaxGrpcProperties.getDefaultApiClientHeaderPattern()));
  }

  @Test
  public void getModelExceptionTest() throws Exception {
    StatusRuntimeException exception = new StatusRuntimeException(io.grpc.Status.INVALID_ARGUMENT);
    mockModelService.addException(exception);

    try {
      String projectId = "projectId-894832108";
      String datasetId = "datasetId-345342029";
      String modelId = "modelId1226956324";
      client.getModel(projectId, datasetId, modelId);
      Assert.fail("No exception raised");
    } catch (InvalidArgumentException e) {
      // Expected exception.
    }
  }

  @Test
  public void listModelsTest() throws Exception {
    ModelProto.ListModelsResponse expectedResponse =
        ModelProto.ListModelsResponse.newBuilder()
            .addAllModels(new ArrayList<ModelProto.Model>())
            .setNextPageToken("nextPageToken-1386094857")
            .build();
    mockModelService.addResponse(expectedResponse);

    String projectId = "projectId-894832108";
    String datasetId = "datasetId-345342029";
    UInt32Value maxResults = UInt32Value.newBuilder().build();

    ModelProto.ListModelsResponse actualResponse =
        client.listModels(projectId, datasetId, maxResults);
    Assert.assertEquals(expectedResponse, actualResponse);

    List<AbstractMessage> actualRequests = mockModelService.getRequests();
    Assert.assertEquals(1, actualRequests.size());
    ModelProto.ListModelsRequest actualRequest =
        ((ModelProto.ListModelsRequest) actualRequests.get(0));

    Assert.assertEquals(projectId, actualRequest.getProjectId());
    Assert.assertEquals(datasetId, actualRequest.getDatasetId());
    Assert.assertEquals(maxResults, actualRequest.getMaxResults());
    Assert.assertTrue(
        channelProvider.isHeaderSent(
            ApiClientHeaderProvider.getDefaultApiClientHeaderKey(),
            GaxGrpcProperties.getDefaultApiClientHeaderPattern()));
  }

  @Test
  public void listModelsExceptionTest() throws Exception {
    StatusRuntimeException exception = new StatusRuntimeException(io.grpc.Status.INVALID_ARGUMENT);
    mockModelService.addException(exception);

    try {
      String projectId = "projectId-894832108";
      String datasetId = "datasetId-345342029";
      UInt32Value maxResults = UInt32Value.newBuilder().build();
      client.listModels(projectId, datasetId, maxResults);
      Assert.fail("No exception raised");
    } catch (InvalidArgumentException e) {
      // Expected exception.
    }
  }

  @Test
  public void patchModelTest() throws Exception {
    ModelProto.Model expectedResponse =
        ModelProto.Model.newBuilder()
            .setEtag("etag3123477")
            .setModelReference(ModelReferenceProto.ModelReference.newBuilder().build())
            .setCreationTime(1932333101)
            .setLastModifiedTime(-671513446)
            .setDescription("description-1724546052")
            .setFriendlyName("friendlyName461933014")
            .putAllLabels(new HashMap<String, String>())
            .setExpirationTime(767170141)
            .setLocation("location1901043637")
            .setEncryptionConfiguration(
                EncryptionConfigProto.EncryptionConfiguration.newBuilder().build())
            .addAllTrainingRuns(new ArrayList<ModelProto.Model.TrainingRun>())
            .addAllFeatureColumns(new ArrayList<StandardSqlProto.StandardSqlField>())
            .addAllLabelColumns(new ArrayList<StandardSqlProto.StandardSqlField>())
            .build();
    mockModelService.addResponse(expectedResponse);

    String projectId = "projectId-894832108";
    String datasetId = "datasetId-345342029";
    String modelId = "modelId1226956324";
    ModelProto.Model model = ModelProto.Model.newBuilder().build();

    ModelProto.Model actualResponse = client.patchModel(projectId, datasetId, modelId, model);
    Assert.assertEquals(expectedResponse, actualResponse);

    List<AbstractMessage> actualRequests = mockModelService.getRequests();
    Assert.assertEquals(1, actualRequests.size());
    ModelProto.PatchModelRequest actualRequest =
        ((ModelProto.PatchModelRequest) actualRequests.get(0));

    Assert.assertEquals(projectId, actualRequest.getProjectId());
    Assert.assertEquals(datasetId, actualRequest.getDatasetId());
    Assert.assertEquals(modelId, actualRequest.getModelId());
    Assert.assertEquals(model, actualRequest.getModel());
    Assert.assertTrue(
        channelProvider.isHeaderSent(
            ApiClientHeaderProvider.getDefaultApiClientHeaderKey(),
            GaxGrpcProperties.getDefaultApiClientHeaderPattern()));
  }

  @Test
  public void patchModelExceptionTest() throws Exception {
    StatusRuntimeException exception = new StatusRuntimeException(io.grpc.Status.INVALID_ARGUMENT);
    mockModelService.addException(exception);

    try {
      String projectId = "projectId-894832108";
      String datasetId = "datasetId-345342029";
      String modelId = "modelId1226956324";
      ModelProto.Model model = ModelProto.Model.newBuilder().build();
      client.patchModel(projectId, datasetId, modelId, model);
      Assert.fail("No exception raised");
    } catch (InvalidArgumentException e) {
      // Expected exception.
    }
  }

  @Test
  public void deleteModelTest() throws Exception {
    Empty expectedResponse = Empty.newBuilder().build();
    mockModelService.addResponse(expectedResponse);

    String projectId = "projectId-894832108";
    String datasetId = "datasetId-345342029";
    String modelId = "modelId1226956324";

    client.deleteModel(projectId, datasetId, modelId);

    List<AbstractMessage> actualRequests = mockModelService.getRequests();
    Assert.assertEquals(1, actualRequests.size());
    ModelProto.DeleteModelRequest actualRequest =
        ((ModelProto.DeleteModelRequest) actualRequests.get(0));

    Assert.assertEquals(projectId, actualRequest.getProjectId());
    Assert.assertEquals(datasetId, actualRequest.getDatasetId());
    Assert.assertEquals(modelId, actualRequest.getModelId());
    Assert.assertTrue(
        channelProvider.isHeaderSent(
            ApiClientHeaderProvider.getDefaultApiClientHeaderKey(),
            GaxGrpcProperties.getDefaultApiClientHeaderPattern()));
  }

  @Test
  public void deleteModelExceptionTest() throws Exception {
    StatusRuntimeException exception = new StatusRuntimeException(io.grpc.Status.INVALID_ARGUMENT);
    mockModelService.addException(exception);

    try {
      String projectId = "projectId-894832108";
      String datasetId = "datasetId-345342029";
      String modelId = "modelId1226956324";
      client.deleteModel(projectId, datasetId, modelId);
      Assert.fail("No exception raised");
    } catch (InvalidArgumentException e) {
      // Expected exception.
    }
  }
}
